service/history/api/respondactivitytaskcanceled/api.go

```diff
@@ -109,6 +112,18 @@ func Invoke(
 				return nil, consts.ErrActivityTaskNotCancelRequested
 			}
 
+			// We fabricate a started event only when the activity is not started yet and
+			// we need to force cancel an activity
+			if isCalledCompleteActivityById && ai.GetRetryLastFailure() != nil {
+				_, err := mutableState.AddActivityTaskStartedEvent(ai, scheduledEventID,
+					uuid.New().String(),
```

Comment:
Can this be empty? It seems better to leave empty than to fabricate it.

Reply:
Will check, if it could be empty then we could put it as an empty, or shall we put a place holder ID here to present such event is a fabricated one?

Reply:
I'd just leave it empty if possible, otherwise use a placeholder like `fabricated-start-event`.